### ISSUE TITLE
[Help with Edits] Adding Tagline to <variable name="Label_56560_InfoLine">

### ISSUE THREAD

**Issue by unkn0wn234** [jurialmunkey/skin.arctic.horizon#381]
Hi jurial first off id like to say if this isnt the appropriate place to ask for help on this let me know I don't know anywhere else to ask and it's your skin, if it's not something you'd want to be bothered with just let me know and I'll close and refrain from asking further.
After looking at other peoples edits on this skin I thought I would try and make some for myself, I enlarged the clearlogo and moved the position on the Fullscreen widget and was able to change that easily.

So where Im stumped currently is this: I edited the Infoline in Includes_Labels to remove the Rated "x", and Ends at "x" lines. Worked successfully around the board (on the fullscreen widget and on widgets with info enabled). However I tried to add the Tagline using the info given in Detailed Items on the TMDBH wiki, and it shows up on widgets with info enabled, but not on the fullscreen widgets. If I press more info on a fullscreen widget and press down into cast and back out, the Tagline will appear on the fullscreen widget although only temporarily.

Would you mind taking a look at the edit I made and letting me know if I did something wrong and if I need to add this edit in more than one place to get it to show on the Fullscreen widget consistently?

![screenshot00003](https://user-images.githubusercontent.com/52363082/111008091-5750a700-835e-11eb-8e33-089775ca1ce4.png)
![screenshot00004](https://user-images.githubusercontent.com/52363082/111008094-57e93d80-835e-11eb-945f-3f67bda4a690.png)
![screenshot00002](https://user-images.githubusercontent.com/52363082/111008095-5881d400-835e-11eb-9579-aa1d7824f04f.png)

**Comment by unkn0wn234** [comment]
https://github.com/jurialmunkey/skin.arctic.horizon/compare/matrix...unkn0wn234:matrix

**Comment by jurialmunkey** [comment]
ListItem by itself refers to the currently focused item. The fullscreen widget never gets focus itself (only the buttons get focus which then control the widget which is the fanart in the background). So referencing the ListItem directly will never work. 

You need to reference the container ID but because each widget has a unique ID that gets difficult. That's why all the info in the infoline references 56560 instead. That is a hidden container which uses a series of skinshortcuts templates tricks to pull in the info for the current widget so that there is standardised ID to reference.

The definition for what is available in 56560 is in Includes_Object.xml
https://github.com/jurialmunkey/skin.arctic.horizon/blob/matrix/1080i/Includes_Object.xml#L3-L56

You need to add a definition for tagline there and then reference 56560 like all the other info labels do. It should be fairly self explanatory in terms of how to add an extra line - just follow the pattern of the other lines.

**Comment by unkn0wn234** [comment]
> ListItem by itself refers to the currently focused item. The fullscreen widget never gets focus itself (only the buttons get focus which then control the widget which is the fanart in the background). So referencing the ListItem directly will never work.
> 
> You need to reference the container ID but because each widget has a unique ID that gets difficult. That's why all the info in the infoline references 56560 instead. That is a hidden container which uses a series of skinshortcuts templates tricks to pull in the info for the current widget so that there is standardised ID to reference.
> 
> The definition for what is available in 56560 is in Includes_Object.xml
> https://github.com/jurialmunkey/skin.arctic.horizon/blob/matrix/1080i/Includes_Object.xml#L3-L56
> 
> You need to add a definition for tagline there and then reference 56560 like all the other info labels do. It should be fairly self explanatory in terms of how to add an extra line - just follow the pattern of the other lines.

Thank you! Im gonna mess with it, also is there a way to change the cutoff length for the Info line? Sometimes with a longer info tag it gets cut off. 

I would ask if theres a way to exclude tagline if it exceeds a certain length but i doubt thats possible and if it is, would take more skill than what I currently have

**Comment by unkn0wn234** [comment]
Oh Also in the More Information view, it doesn't show the Tagline when there are Awards, it will show 7 lines, even tho theres space in the bottom right for an 8th any thoughts? Ill try and get a screencap later when I get on my computer

**Comment by jurialmunkey** [comment]
> is there a way to change the cutoff length for the Info line? Sometimes with a longer info tag it gets cut off.

`Includes_Viewtype.xml`
```xml
<include name="Viewtype_Info_Right"><right>600</right></include>
<include name="Viewtype_Info_Right_FixedHome"><right>560</right></include>
```

> I would ask if theres a way to exclude tagline if it exceeds a certain length but i doubt thats possible and if it is, would take more skill than what I currently have

Nope. Skinning engine doesn't provide any way to read width/length of controls/text.

> Oh Also in the More Information view, it doesn't show the Tagline when there are Awards, it will show 7 lines, even tho theres space in the bottom right for an 8th any thoughts? Ill try and get a screencap later when I get on my computer

Not if there's a filename and codec info:

![Honeyview_screenshot00022](https://user-images.githubusercontent.com/4618719/111016670-59b90c80-8403-11eb-80bb-0df8dae8d171.jpg)

**Comment by unkn0wn234** [comment]
> > is there a way to change the cutoff length for the Info line? Sometimes with a longer info tag it gets cut off.
> 
> `Includes_Viewtype.xml`
> 
> ```
> <include name="Viewtype_Info_Right"><right>600</right></include>
> <include name="Viewtype_Info_Right_FixedHome"><right>560</right></include>
> ```
> 
> > I would ask if theres a way to exclude tagline if it exceeds a certain length but i doubt thats possible and if it is, would take more skill than what I currently have
> 
> Nope. Skinning engine doesn't provide any way to read width/length of controls/text.
> 
> > Oh Also in the More Information view, it doesn't show the Tagline when there are Awards, it will show 7 lines, even tho theres space in the bottom right for an 8th any thoughts? Ill try and get a screencap later when I get on my computer
> 
> Not if there's a filename and codec info:
> 
> ![Honeyview_screenshot00022](https://user-images.githubusercontent.com/4618719/111016670-59b90c80-8403-11eb-80bb-0df8dae8d171.jpg)

Thank you for the first part I will play with it!, and regarding the second figured as much but worth asking, and regarding the third. For me it happens when i press info from a widget, so where the Filename is, its empty.

So I do have another question, thank you very much for the responses and help thus far though I appreciate it a lot! So in that last image you just sent of Fant. Beasts, Ill refer to it as the info dialog view.

I edited the info dialog view to show Listitem.Premiered rather than year so when Im on the home screen and see a movie with 2021, I click information, it will then show the full premiere date. I achieved that how I wanted to, but the problem for me that I noticed is, if I go into the TMDBHelper or any other Addon and go to movies. The info line matches the info dialog view, and not the one from the home view. 

Is this by design? If so is it open to change? I would think having the info dialogs info line seperate from the rest would be ideal so that you could have certain things in the addon and home view, and when u press more info, the info dialog info line shows more/different stuff.

**Comment by jurialmunkey** [comment]
You'll see in this variable that there are two different infoline variables:
https://github.com/jurialmunkey/skin.arctic.horizon/blob/matrix/1080i/Includes_Labels.xml#L83-L86

Basically when 56560 is available, it will use the variable that references that container otherwise it will use the simpler variable which references the listitem directly. 56560 is available in places like the home screen where there are multiple container IDs to make it possible to retrieve the details. In other places where there is a default container such as in the library or the info dialog, we don't need to use 56560 so it is skipped as it is much quicker to retrieve info directly from the list item.

The intent is that the info line is standardised across the whole skin and appears the same no matter where you are. I'm a big proponent of ensuring that information is always in the same place on screen as much as possible - it might not seem like a big deal initially but after months of use your brain gets used to information being in a certain spot. That means immediately when looking for a piece of information your eyes will go to a particular spot on screen -- when that spot changes from situation to situation, it means your eyes/brain have to work harder to find the information which is fatiguing and means the interface becomes a hurdle to navigate. I find even the fact that the poster in the info dialog pushes over the position of the info to be annoying but it is a necessary compromise because a right aligned poster is more fatiguing due to LTR text needing to be left aligned.

**Comment by unkn0wn234** [comment]
> You'll see in this variable that there are two different infoline variables:
> https://github.com/jurialmunkey/skin.arctic.horizon/blob/matrix/1080i/Includes_Labels.xml#L83-L86
> 
> Basically when 56560 is available, it will use the variable that references that container otherwise it will use the simpler variable which references the listitem directly. 56560 is available in places like the home screen where there are multiple container IDs to make it possible to retrieve the details. In other places where there is a default container such as in the library or the info dialog, we don't need to use 56560 so it is skipped as it is much quicker to retrieve info directly from the list item.
> 
> The intent is that the info line is standardised across the whole skin and appears the same no matter where you are. I'm a big proponent of ensuring that information is always in the same place on screen as much as possible - it might not seem like a big deal initially but after months of use your brain gets used to information being in a certain spot. That means immediately when looking for a piece of information your eyes will go to a particular spot on screen -- when that spot changes from situation to situation, it means your eyes/brain have to work harder to find the information which is fatiguing and means the interface becomes a hurdle to navigate. I find even the fact that the poster in the info dialog pushes over the position of the info to be annoying but it is a necessary compromise because a right aligned poster is more fatiguing due to LTR text needing to be left aligned.

Ah thanks for explaining but I haven't fully grasped it. I understand you don't want to make it this way for the reasons above, but does this mean it is possible if I wanted to make the change for myself, to make addon views/library use the 56560 view as in make it match the infoline on the home screen?

**Comment by unkn0wn234** [comment]
> > You'll see in this variable that there are two different infoline variables:
> > https://github.com/jurialmunkey/skin.arctic.horizon/blob/matrix/1080i/Includes_Labels.xml#L83-L86
> > Basically when 56560 is available, it will use the variable that references that container otherwise it will use the simpler variable which references the listitem directly. 56560 is available in places like the home screen where there are multiple container IDs to make it possible to retrieve the details. In other places where there is a default container such as in the library or the info dialog, we don't need to use 56560 so it is skipped as it is much quicker to retrieve info directly from the list item.
> > The intent is that the info line is standardised across the whole skin and appears the same no matter where you are. I'm a big proponent of ensuring that information is always in the same place on screen as much as possible - it might not seem like a big deal initially but after months of use your brain gets used to information being in a certain spot. That means immediately when looking for a piece of information your eyes will go to a particular spot on screen -- when that spot changes from situation to situation, it means your eyes/brain have to work harder to find the information which is fatiguing and means the interface becomes a hurdle to navigate. I find even the fact that the poster in the info dialog pushes over the position of the info to be annoying but it is a necessary compromise because a right aligned poster is more fatiguing due to LTR text needing to be left aligned.
> 
> Ah thanks for explaining but I haven't fully grasped it. I understand you don't want to make it this way for the reasons above, but does this mean it is possible if I wanted to make the change for myself, to force the addon views/library use the 56560 view as in make it match the infoline on the home screen?

**Comment by jurialmunkey** [comment]
> Ah thanks for explaining but I haven't fully grasped it. I understand you don't want to make it this way for the reasons above, but does this mean it is possible if I wanted to make the change for myself, to make addon views/library use the 56560 view as in make it match the infoline on the home screen?

You just need to edit both variables to match:

Edit `Label_56560_InfoLine` using `$INFO[Container(56560).ListItem.Property(tagline)]` (or whatever you called the property in Includes_Object).
https://github.com/jurialmunkey/skin.arctic.horizon/blob/matrix/1080i/Includes_Labels.xml#L93

Then in `Label_InfoLine` make the same edits but use `$INFO[ListItem.Tagline]` instead
https://github.com/jurialmunkey/skin.arctic.horizon/blob/matrix/1080i/Includes_Labels.xml#L146

**Comment by unkn0wn234** [comment]
> > Ah thanks for explaining but I haven't fully grasped it. I understand you don't want to make it this way for the reasons above, but does this mean it is possible if I wanted to make the change for myself, to make addon views/library use the 56560 view as in make it match the infoline on the home screen?
> 
> You just need to edit both variables to match:
> 
> Edit `Label_56560_InfoLine` using `$INFO[Container(56560).ListItem.Property(tagline)]` (or whatever you called the property in Includes_Object).
> https://github.com/jurialmunkey/skin.arctic.horizon/blob/matrix/1080i/Includes_Labels.xml#L93
> 
> Then in `Label_InfoLine` make the same edits but use `$INFO[ListItem.Tagline]` instead
> https://github.com/jurialmunkey/skin.arctic.horizon/blob/matrix/1080i/Includes_Labels.xml#L146

Sorry I think I misphrased the question unless this is actually what I'm looking for. If I do this it _would_ match with the 56560 version, but what I am trying to do is make it so the info dialog uses the `Label_Infoline` only while everything else uses `Label_56560_InfoLine` so like the home screen AND addons. Is there somewhere else in an xml that tells addons/library content to pull from `Label_InfoLine` that I could change to pull from `Label_56560_InfoLine` ? 

Basically I only want `Label_InfoLine` to be used for the info dialog in my changes I don't mind the extra load times or inconsistency.

**Comment by jurialmunkey** [comment]
You need to add the 56560 container to myvideonav. It can't use it if it doesn't exist. That's the whole point of the parent variable - if container 56560 exists it gets used otherwise it uses the the direct list item one.

**Comment by unkn0wn234** [comment]
> You need to add the 56560 container to myvideonav. It can't use it if it doesn't exist. That's the whole point of the parent variable - if container 56560 exists it gets used otherwise it uses the the direct list item one.

This is what I was looking for thanks I will try and add it to my videonav, Im assuming once I do this, the addons/library content will default to showing the 56560 container version for the info line? and that i could reference the home widget/fullscreen widget version and just copy it? Ill probably figure it out as i look into it anyway

**Comment by unkn0wn234** [comment]
> > You need to add the 56560 container to myvideonav. It can't use it if it doesn't exist. That's the whole point of the parent variable - if container 56560 exists it gets used otherwise it uses the the direct list item one.
> 
> This is what I was looking for thanks I will try and add it to my videonav, Im assuming once I do this, the addons/library content will default to showing the 56560 container version for the info line? and that i could reference the home widget/fullscreen widget version and just copy it? Ill probably figure it out as i look into it anyway

Sorry I am not really understanding where/what Im supposed to do, but heres what I think Im supposed to do could you correct me? 

I added `<param name="info" value="Label_56560_InfoLine" />`  to the  `<include content="Dialog_Menu">`  section is this correct? 

Actually I think it should be this `<param name="info" value="$VAR[Label_56560_InfoLine]" />` ?

Well none of these worked so far lol Ill keep trying I think its actually more complex than me just adding a param to Dialog_Menu

Ok I think I might be getting somehwere I added this `<param name="Label_56560_InfoLine" value=""Control.IsVisible(56560)">$VAR[Label_56560_InfoLine]" />` but it makes a black screen once I go into addons, I removed the extra "" and now it loads up but its empty, no black screen. I believe I remember u saying the 56560 container is a hidden container thats just there, how would I make it hidden in the MyVideoNav? I dont think its as simple as Control.IsNotVisible, theres a section called  but im not sure how to go about adding it in there if thats the correct route

**Comment by jurialmunkey** [comment]
None of that will work. You should read how variables, includes and params work in the skinning manual because you are mixing syntax and just randomly copy pasting stuff is not going to work. 
https://kodi.wiki/view/Skinning_Manual#Includes

Also Dialog_Menu has absolutely nothing to do with any of this -- it is the context menu dialog shown for the "Similar" button.

As you see in this variable, if the 56560 control is visible it will use the 56560 variable for the infoline otherwise it uses the standard one.
```xml
<variable name="Label_Home_InfoLine">
    <value condition="Control.IsVisible(56560)">$VAR[Label_56560_InfoLine]</value>
    <value>$VAR[Label_InfoLine]</value>
</variable>
```

The issue is that control 56560 is not visible for the viewtypes that you specify. You can see that here:
https://github.com/jurialmunkey/skin.arctic.horizon/blob/matrix/1080i/Includes_Viewtype.xml#L31-L48

As you can see in the visible line of that container, it is only visible for views 53, 58, 59.
```xml
<visible>Control.IsVisible(53) | Control.IsVisible(58) | Control.IsVisible(59)</visible>
```

Those viewtypes are the "integrated" views for each aspect type (e.g. 53 is the "seasons" view that shows seasons as buttons with landscape underneath and 58/59 are the same but for posters and square aspects). The reason those views use 56560 is because the bit underneath the buttons essentially counts as a widget and so can't be referenced directly using listitem label.

If you look in the content tags of that 56560 control you will see that it has special content for each of those views that references a different container ID. For example, the one for 53 actually references 530. That's because the _main_ view ID for seasons is 53 (the buttons part) but the underneath widget has the ID 530. So basically when the button part is not focused then 56560 will be used.

Further, the main grouplist showing the info line in the library uses the default variable (standard info line), so you need to change that. You can see the info grouplist for viewtypes is defined here:
https://github.com/jurialmunkey/skin.arctic.horizon/blob/matrix/1080i/Includes_Viewtype.xml#L85-L94

When you look at the definition for `View_Info_Grouplist` include you can see in the InfoLine part that it passes through the param from the parent include:
https://github.com/jurialmunkey/skin.arctic.horizon/blob/matrix/1080i/Includes_View_Info.xml#L44-L47

Since the parent include does not define that param, the infoline object instead uses the default defined param, which is defined here:
https://github.com/jurialmunkey/skin.arctic.horizon/blob/matrix/1080i/Includes_Object.xml#L661

To get it to use something other than the default, you need to pass through the relevant param to the parent `View_Info_Grouplist` include (which then passes through the same param to the infoline object include).


To sum:

- In viewtypes, you need to change the 56560 container to be visible when you want it to be used
- You need to add an appropriate content include to that container to reference the default container and make it visible when you want it used. You can reference the default container by using `ListItem.xxxxx` without the `Container(ID).` prefix
- You need to update the infoline for the info grouplist to reference the `Label_Home_InfoLine` rather than `Label_InfoLine`. There are multiple different ways to do this but since that include is used in a lot of places, the best thing to do would be to pass the param to the parent include in the viewtypes includes. For an example, look at the 1110 hub code to see how the hub changes which variable the include uses: https://github.com/jurialmunkey/skin.arctic.horizon/blob/matrix/1080i/Custom_1110_Hub.xml#L61

>  I dont think its as simple as Control.IsNotVisible, theres a section called but im not sure how to go about adding it in there if thats the correct route

Read about conditional visibility in the skinning manual:
https://kodi.wiki/view/Skinning_Manual#Conditional_Visibility

You can use `!` to specify a NOT conditional.

**Comment by unkn0wn234** [comment]
> None of that will work. You should read how variables, includes and params work in the skinning manual because you are mixing syntax and just randomly copy pasting stuff is not going to work.
> https://kodi.wiki/view/Skinning_Manual#Includes
> 
> Also Dialog_Menu has absolutely nothing to do with any of this -- it is the context menu dialog shown for the "Similar" button.
> 
> As you see in this variable, if the 56560 control is visible it will use the 56560 variable for the infoline otherwise it uses the standard one.
> 
> ```
> <variable name="Label_Home_InfoLine">
>     <value condition="Control.IsVisible(56560)">$VAR[Label_56560_InfoLine]</value>
>     <value>$VAR[Label_InfoLine]</value>
> </variable>
> ```
> 
> The issue is that control 56560 is not visible for the viewtypes that you specify. You can see that here:
> https://github.com/jurialmunkey/skin.arctic.horizon/blob/matrix/1080i/Includes_Viewtype.xml#L31-L48
> 
> As you can see in the visible line of that container, it is only visible for views 53, 58, 59.
> 
> ```
> <visible>Control.IsVisible(53) | Control.IsVisible(58) | Control.IsVisible(59)</visible>
> ```
> 
> Those viewtypes are the "integrated" views for each aspect type (e.g. 53 is the "seasons" view that shows seasons as buttons with landscape underneath and 58/59 are the same but for posters and square aspects). The reason those views use 56560 is because the bit underneath the buttons essentially counts as a widget and so can't be referenced directly using listitem label.
> 
> If you look in the content tags of that 56560 control you will see that it has special content for each of those views that references a different container ID. For example, the one for 53 actually references 530. That's because the _main_ view ID for seasons is 53 (the buttons part) but the underneath widget has the ID 530. So basically when the button part is not focused then 56560 will be used.
> 
> Further, the main grouplist showing the info line in the library uses the default variable (standard info line), so you need to change that. You can see the info grouplist for viewtypes is defined here:
> https://github.com/jurialmunkey/skin.arctic.horizon/blob/matrix/1080i/Includes_Viewtype.xml#L85-L94
> 
> When you look at the definition for `View_Info_Grouplist` include you can see in the InfoLine part that it passes through the param from the parent include:
> https://github.com/jurialmunkey/skin.arctic.horizon/blob/matrix/1080i/Includes_View_Info.xml#L44-L47
> 
> Since the parent include does not define that param, the infoline object instead uses the default defined param, which is defined here:
> https://github.com/jurialmunkey/skin.arctic.horizon/blob/matrix/1080i/Includes_Object.xml#L661
> 
> To get it to use something other than the default, you need to pass through the relevant param to the parent `View_Info_Grouplist` include (which then passes through the same param to the infoline object include).
> 
> To sum:
> 
> * In viewtypes, you need to change the 56560 container to be visible when you want it to be used
> * You need to add an appropriate content include to that container to reference the default container and make it visible when you want it used. You can reference the default container by using `ListItem.xxxxx` without the `Container(ID).` prefix
> * You need to update the infoline for the info grouplist to reference the `Label_Home_InfoLine` rather than `Label_InfoLine`. There are multiple different ways to do this but since that include is used in a lot of places, the best thing to do would be to pass the param to the parent include in the viewtypes includes. For an example, look at the 1110 hub code to see how the hub changes which variable the include uses: https://github.com/jurialmunkey/skin.arctic.horizon/blob/matrix/1080i/Custom_1110_Hub.xml#L61
> 
> > I dont think its as simple as Control.IsNotVisible, theres a section called but im not sure how to go about adding it in there if thats the correct route
> 
> Read about conditional visibility in the skinning manual:
> https://kodi.wiki/view/Skinning_Manual#Conditional_Visibility
> 
> You can use `!` to specify a NOT conditional.

Thanks for the thorough explanation I follow about half of it but I don't think that it's realistic for me to achieve with the minimal knowledge I have on the subject so I think the best route of action is just to go with your original suggestion and make the infolines match universally. I do have another question that I hope will be simpler to go about, where can I change the content that shows in the information dialog (the bottom part in the picture u sent: Writer, Director, Tagline, Revenue/Budget, etc.
I believe I saw it originally in Includes_Labels

Once again though thank you so much for taking the time to lay things out for me

**Comment by jurialmunkey** [comment]
https://github.com/jurialmunkey/skin.arctic.horizon/blob/matrix/1080i/Includes_DialogVideoInfo.xml#L487-L642

**Comment by unkn0wn234** [comment]
Hi jurial id rather not open a new post just to ask this so just going to ask here and if the answer is no, just wondering your thoughts on why not as I was really intrigued by your explanation on the previous topic/issue here

> The intent is that the info line is standardized across the whole skin and appears the same no matter where you are. I'm a big proponent of ensuring that information is always in the same place on screen as much as possible - it might not seem like a big deal initially but after months of use your brain gets used to information being in a certain spot. That means immediately when looking for a piece of information your eyes will go to a particular spot on screen -- when that spot changes from situation to situation, it means your eyes/brain have to work harder to find the information which is fatiguing and means the interface becomes a hurdle to navigate. I find even the fact that the poster in the info dialog pushes over the position of the info to be annoying but it is a necessary compromise because a right aligned poster is more fatiguing due to LTR text needing to be left aligned.

After a lot of messing w the infoline i realized the taglines sometimes are too long, and would cut off the info line which could be fixed by editing the cutoff length on the home screen but not in the info dialog if i wanted to make it universal.
Would you consider adding a line under the title for the Tagline? Like with Netfilx/Titan Bingie/Mod does here
![vPe2ErW](https://user-images.githubusercontent.com/52363082/111246889-7741ce00-85dd-11eb-8263-87fe82ac0425.jpg)

Regarding Shows, they use "Watch Season "x" now" with x being either the season you're currently on, or either 1 or the latest season out (Edit: Just noticed a lot of shows have Taglines anyway so would be nice to have them in place there too then actually)

![maxresdefault](https://user-images.githubusercontent.com/52363082/111247126-dbfd2880-85dd-11eb-9e5a-81e62daea6eb.jpg)

I made a shitty mockup for AH of how it would look but again would like to hear your thoughts on the possibility
![screenshot00002](https://user-images.githubusercontent.com/52363082/111247255-0949d680-85de-11eb-851c-73333a9b5645.png)

**Comment by jurialmunkey** [comment]
Looks way too cluttered. Whitespace is really important in design and the skin has a set of padding requirements to ensure that elements can breathe.

Here's a good article on the importance of whitespace
https://medium.com/wayfair-design/more-padding-please-b95e19422acc

**Comment by unkn0wn234** [comment]
> Looks way too cluttered. Whitespace is really important in design and the skin has a set of padding requirements to ensure that elements can breathe.
> 
> Here's a good article on the importance of whitespace
> https://medium.com/wayfair-design/more-padding-please-b95e19422acc

So do you think with even more space apart it would still be too much information on it? Thanks for the article and explanation, Ill have to make do just putting it in the info line or not putting it in at all

**Comment by jurialmunkey** [comment]
I think probably the only real way to fit it whilst keeping a sense of space to the elements would be to do something similar to how the episode title is included in bold with the plot section (if you open up a season of a show you will see what I mean).

In the vertical space available I just don't think it's possible to fit another line of information whilst keeping the current view dimensions and not making it cluttered. It would need to replace something else (eg info line or ratings)

**Comment by unkn0wn234** [comment]
> I think probably the only real way to fit it whilst keeping a sense of space to the elements would be to do something similar to how the episode title is included in bold with the plot section (if you open up a season of a show you will see what I mean).

I had a look and yea this would be perfect. Would you consider implementing/adding an option for this? Or would it be not too difficult to add myself?

And also after looking more I actually think I figured out the perfect solution if you could show me how to edit this/these or consider adding it.
![screenshot00001](https://user-images.githubusercontent.com/52363082/111427646-cdda0580-86cc-11eb-9709-9da2c470637b.png)
![screenshot00000](https://user-images.githubusercontent.com/52363082/111427648-ce729c00-86cc-11eb-96b8-969f12ece7b2.png)

So on the Home Screen on top of the widget it says the name of where its pulled from followed by the movie name. "Popular | Coming 2 America" would it be possible to add the Tagline to it? so it would be "Popular | Coming 2 America | Tagline"

I was going to say replace the title with the tagline but then on widgets with no info enabled it would replace the only place u can see the name. 

I did notice though inside Widgets it says "Sort by: x" instead of the movie/show name though. So it wouldnt be useful there but I think having the Title there would be better, or replace it with the tagline. 

And I say it would be better not just for the reasons I said above bc when u have poster wall for Seasons it would be much better to see the Season number than "Sorted by" like here.
![screenshot00002](https://user-images.githubusercontent.com/52363082/111428293-ad5e7b00-86cd-11eb-804a-257bf605cb0f.png)

**Comment by jurialmunkey** [comment]
Widget label is defined here:
https://github.com/jurialmunkey/skin.arctic.horizon/blob/matrix/1080i/Includes_View.xml#L99-L118

First label control is the bold left part, second label is the divider, and third label is the right side label.

Just duplicate the second and third control and tweak as needed to get the tagline. Should be fairly self explanatory.

You can see in the third label there is also a `<label>$PARAM[altlabel]</label>` part. That's how the viewtypes override the right side. For instance, you can see the override here (in the library the control with ID 3 and 4 are the sorting labels which is why it references them).
https://github.com/jurialmunkey/skin.arctic.horizon/blob/matrix/1080i/Includes_View_52_Posters.xml#L98

**Comment by unkn0wn234** [comment]
> Widget label is defined here:
> https://github.com/jurialmunkey/skin.arctic.horizon/blob/matrix/1080i/Includes_View.xml#L99-L118
> 
> First label control is the bold left part, second label is the divider, and third label is the right side label.
> 
> Just duplicate the second and third control and tweak as needed to get the tagline. Should be fairly self explanatory.
> 
> You can see in the third label there is also a `<label>$PARAM[altlabel]</label>` part. That's how the viewtypes override the right side. For instance, you can see the override here (in the library the control with ID 3 and 4 are the sorting labels which is why it references them).
> https://github.com/jurialmunkey/skin.arctic.horizon/blob/matrix/1080i/Includes_View_52_Posters.xml#L98

![screenshot00003](https://user-images.githubusercontent.com/52363082/111443692-020af180-86e0-11eb-841d-e51c24c7798c.png)
Thank you so much jurial I think this is the best outcome. It's great, the Tagline still shows up and it doesn't look like it takes up too much space. I removed the altlabel and added the Tagline next. I do have a few more questions to bother you with lol but really thank you so much.

![screenshot00005](https://user-images.githubusercontent.com/52363082/111444018-53b37c00-86e0-11eb-91e0-11bdd8a357be.png)
1. So when there is no Tagline, it will show a | after the Title instead of nothing, is there no way to  make it not show up if no Tagline is detected? If it can't be done I really could care less bc I'm happy with how it turned out regardless being able to get the Tagline in without messing up the open space.

![screenshot00006](https://user-images.githubusercontent.com/52363082/111444680-0257bc80-86e1-11eb-80bf-7465c5713930.png)
2. For now I just removed the altlabel to get it to show the show name and season number, but is it possible to make the altlabel of "Sort by" only show up for certain dbtypes like addons? Again not a big deal but i am just curious if its somewhat easily doable.

3. So now that I have the Tagline on the Widget Label I do have one last question but I think its probably unlikely. Is there any way to make it so the Fullscreen widget uses the Tagline for the infoline but not the rest of the widgets on the homescreen? If its not possible then so be it a small price to pay for being able to customize it how i wanted to

**Comment by unkn0wn234** [comment]
Hey jurial so Im noticing that my edited InfoLine doesnt show up in other addons that I thought would pull from Label_InfoLine is there something else i can or am supposed to do/have i done something incorrectly? Or is it that the InfoLine for addons is completely dependent on that addon and Im kinda SOL?

**Comment by unkn0wn234** [comment]
> > Widget label is defined here:
> > https://github.com/jurialmunkey/skin.arctic.horizon/blob/matrix/1080i/Includes_View.xml#L99-L118
> > First label control is the bold left part, second label is the divider, and third label is the right side label.
> > Just duplicate the second and third control and tweak as needed to get the tagline. Should be fairly self explanatory.
> > You can see in the third label there is also a `<label>$PARAM[altlabel]</label>` part. That's how the viewtypes override the right side. For instance, you can see the override here (in the library the control with ID 3 and 4 are the sorting labels which is why it references them).
> > https://github.com/jurialmunkey/skin.arctic.horizon/blob/matrix/1080i/Includes_View_52_Posters.xml#L98
> 
> ![screenshot00003](https://user-images.githubusercontent.com/52363082/111443692-020af180-86e0-11eb-841d-e51c24c7798c.png)
> Thank you so much jurial I think this is the best outcome. It's great, the Tagline still shows up and it doesn't look like it takes up too much space. I removed the altlabel and added the Tagline next. I do have a few more questions to bother you with lol but really thank you so much.
> 
> ![screenshot00005](https://user-images.githubusercontent.com/52363082/111444018-53b37c00-86e0-11eb-91e0-11bdd8a357be.png)
> 
> 1. So when there is no Tagline, it will show a | after the Title instead of nothing, is there no way to  make it not show up if no Tagline is detected? If it can't be done I really could care less bc I'm happy with how it turned out regardless being able to get the Tagline in without messing up the open space.
> 
> ![screenshot00006](https://user-images.githubusercontent.com/52363082/111444680-0257bc80-86e1-11eb-80bf-7465c5713930.png)
> 2. For now I just removed the altlabel to get it to show the show name and season number, but is it possible to make the altlabel of "Sort by" only show up for certain dbtypes like addons? Again not a big deal but i am just curious if its somewhat easily doable.
> 
> 1. So now that I have the Tagline on the Widget Label I do have one last question but I think its probably unlikely. Is there any way to make it so the Fullscreen widget uses the Tagline for the infoline but not the rest of the widgets on the homescreen? If its not possible then so be it a small price to pay for being able to customize it how i wanted to

Hi @jurialmunkey some more help would be appreciated sorry to bother you as I am working on making small changes in my fork of your great skin.
 
Regarding 1. Solved it by just adding the ! variables for visibility so if there isnt a Tagline it wont show, pretty simple.

2. Kinda adding to this question I guess, where is Alt Label defined? As of now I just removed it completely in includes_view, but if its possible I would basically just to make it so that when Kodi detects Seasons it wont use the altlabel like it normally does, but continue to use it as normally elsewhere like in addons, and sections in addons, though again nto a big deal.

3. Eh tbh I just dealt w it idrc but basically the only reason I asked was because I would like Taglines to show up in the Spotlight but no where else, though again not a big deal.

4. I am actually writing here because I would like a bit of advice or just pointers on where to look at. So I am trying to take up what was asked here: [https://github.com/jurialmunkey/skin.arctic.horizon/issues/392] Ive gotten the TMDB Rating added to the Widget Line using the Detailed Items from the TMDBH wiki. However Im not sure how to go about adding the PNGs/icons for the specific Rating. https://github.com/unkn0wn234/skin.arctic.horizonmod/blob/matrix/1080i/Includes_View.xml#L139-L160 this is what I have now to start.

4b. This is a question in relation to above, so in AuraMOD this is already been done, I know you dont maintain AuraMOD, but since it is a fork of your own skin I thought maybe you could point me to where in Aura itself the Widget line details are located because then maybe I could just go there and use it as a reference for what they edited from stock Aura to their version with the ratings in tact.

![0e310c41-a91e-42fb-8a00-83d333e7678c](https://user-images.githubusercontent.com/52363082/116178070-ee8d7600-a6e2-11eb-9265-add609396896.jpg)
![9cff481d-df66-4eb0-903e-060cef891c1d](https://user-images.githubusercontent.com/52363082/116178071-efbea300-a6e2-11eb-8a37-c1bc833ab1e6.jpg)
